conjure/controllers/steps/common.py

```diff
@@ -49,3 +54,61 @@ def parse_title(step):
 
 def run_script(path):
     return run(path, shell=True, stderr=PIPE, stdout=PIPE, env=app.env)
+
+
+def wait_for_steps(steps, message_cb, icon_state=None):
+    """ Waits for post processing steps and return its results
+
+    Arguments:
+    steps: list of steps to run
+    message_cb: log writer
+    icon_state: optionally set an icon state (gui only)
+    """
+
+    info = model_info(app.current_model)
+    # Set our provider type environment var so that it is
+    # exposed in future processing tasks
+    app.env['JUJU_PROVIDERTYPE'] = info['ProviderType']
+
+    results = []
+    steps_queue = deque()
+    for step in steps:
+        if "00_pre.sh" in step \
+           or "00_post-bootstrap.sh" in step \
+           or "00_deploy-done.sh" in step:
```

Comment:
I've seen code for ignoring these non-step scripts in a few places now...
Is there a reason these scripts end up in steps? 
could we name them something else that doesn't get slurped up by the glob? maybe the glob could look for ("step-*.sh") and these could be named something else? step-NN-whatever.sh would still sort correctly, and we could simplify some of this code